### PR TITLE
Cr 2493 non car storage account public access

### PR DESCRIPTION
**Pull request by doronafota**

### CODE REVIEW

File: cloudrail/knowledge/rules/azure/non_context_aware/storage_account_public_access_rule.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    def get_id(self) -> str:
+        return 'non_car_storage_account_public_access'
+
+    def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for storage_account in env_context.storage_accounts:
+            if storage_account.allow_blob_public_access:
+                issues.append(
+                    Issue(
+
```
Comment (Imanuel-Mizrahi) [inline cloudrail/knowledge/rules/azure/non_context_aware/storage_account_public_access_rule.py:20]: Please remove this space

File: tests/knowledge/rules/azure/non_context_aware/test_storage_account_public_access_rule.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+class TestStorageAccountPublicAccessRule(unittest.TestCase):
+    def setUp(self):
+        self.rule = StorageAccountPublicAccessRule()
+
+    @parameterized.expand(
+        [
+            ['storage_account_public_access_disabled', False, False],
+            ['storage_account_public_access_enabled', True, True],
+            ['storage_account_public_access_enabled', None, False],
```
Comment (Imanuel-Mizrahi) [inline tests/knowledge/rules/azure/non_context_aware/test_storage_account_public_access_rule.py:20]: I am not sure you need to add a test case for None. Is there any scenario in which this could be None?

**Review by Imanuel-Mizrahi — CHANGES_REQUESTED** [review]

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/knowledge/context/azure/storage/azure_storage_account.py [hunk 3]
```diff
@@ -12,16 +12,19 @@ class AzureStorageAccount(AzureResource):
             account_replication_type: The replication type of the storage account
             network_rules: The networking rules to allow or deny access from.
             enable_https_traffic_only: A flag indicating if only https traffic is allowed
+            allow_blob_public_access: A flag indicator, True if enable public access to containers and blobs else disable.
     """
 
-    def __init__(self, storage_name: str, account_tier: str, account_replication_type: str, enable_https_traffic_only: bool) -> None:
+    def __init__(self, storage_name: str, account_tier: str, account_replication_type: str,
```
Comment (Tomerva) [inline cloudrail/knowledge/context/azure/storage/azure_storage_account.py:18]: __init__ should not have return type

**Review by Tomerva — APPROVED** [review]
